File: 1/p2.go
package main

import "io/ioutil"
import "fmt"
import "strconv"
import "strings"
import "math"


func iterate (m int) int {
    return int(math.Floor(float64(m) / 3.0)) - 2
}

func fuelForMass (m int) int {
    total := 0
    for m > 0 {
        m = iterate(m)
        if m > 0 {
            total += m
        }
    }
    return total
}

func main () {
    data, _ := ioutil.ReadFile("masses.txt")

    fuel := 0
    for _, s := range strings.Split(string(data), "\n") {
        i, err := strconv.Atoi(s)
        if err == nil {
            fuel += fuelForMass(i)
        }
    }

    fmt.Println("total fuel", fuel)
}

File: 1/p2.py
import math


def iterate(m):
    return int(math.floor(m / 3)) - 2


def fuel_for_mass(m):
    total = 0
    while m > 0:
        m = iterate(m)
        if m > 0:
            total += m
    return total


total_fuel = 0
with open('masses.txt') as masses:
    for l in masses.readlines():
        total_fuel += fuel_for_mass(int(l))

print 'total fuel:', total_fuel
